rustpq/pqcrypto
# PQClean: Update to latest master

Hi Thom,
this PR updates PQClean to latest master and makes the needed changes to the Rust wrappers.
Please find below a short list of changes.

* Add NTRU Prime schemes
* Change Crystals-Dilithium4 to Crystals-Dilithium5
* Disable doc-tests for RainbowV because of stack overflow
* Update HQC to newest schemes
* Disable AVX2 implementation of HQC-RMRS-256 as there is a bug in
"compute_syndromes()" in reed_solomon.c
* Sort the scheme variants alphabetically in implementations.yaml

This PR also fixes issue #17  as I have added a note to the implementations.yaml.
I have also added the compiler flag carry-less multiplication "pclmul" and "sse2" to the AVX2 config for HQC compilation.
Should we summarize all compiler flags under AVX2 or should we create a separate flag for SSE2 and pclmul?
